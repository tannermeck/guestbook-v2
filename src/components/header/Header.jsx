import './header.css'

const Header = () => {
    return (
        <section className='header'>
            <div>
                <img src='https://alchemy-guestbook.netlify.app/static/media/guestbook.b8329dbf.png' style={{height: 50 }}alt='book' />
            </div>
            <div>
                <h3>Sign In Friend</h3>
            </div>
        </section>
    )
}
export default Header;