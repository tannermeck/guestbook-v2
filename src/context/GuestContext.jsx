import { useContext, useState, createContext } from 'react';

const GuestContext = createContext();

const GuestProvider = ({children}) => {
    const [guest, setGuest] = useState('')

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
const useGuest = () => {
    const context = useContext(GuestContext);
    if (context === undefined){
        throw new Error('useGuest hook must be called inside of a GuessContext Provider');
    }
    return context;
}
export { GuestProvider, useGuest}