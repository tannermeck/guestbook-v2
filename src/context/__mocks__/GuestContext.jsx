import { createContext, useState } from "react";

const GuestContext = createContext();

const GuestProvider = ({mockUser, children}) => {
    const [guest, setGuest] = useState(
    mockUser ? 'tanner' : '' )

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
export { GuestProvider, GuestContext }