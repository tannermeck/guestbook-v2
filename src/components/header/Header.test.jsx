import { screen, render } from '@testing-library/react';
import { GuestProvider } from '../../context/GuestContext';
import Header from './Header';

it('should add a header component', async() => {
    const { container } = render(
                            <GuestProvider>
                                <Header />
                            </GuestProvider>)
    const message = await screen.findByText('Sign in Friend!')
    expect(message).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});