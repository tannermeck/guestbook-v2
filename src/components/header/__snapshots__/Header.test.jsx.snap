// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a header component 1`] = `
<div>
  <section
    class="header"
  >
    <div>
      <img
        alt="book"
        src="https://alchemy-guestbook.netlify.app/static/media/guestbook.b8329dbf.png"
        style="height: 50px;"
      />
    </div>
    <div
      class="welcome"
    >
      <h3>
        Sign in Friend!
      </h3>
    </div>
  </section>
</div>
`;
